client/container_create.go

```diff
@@ -61,3 +60,20 @@ func (cli *Client) ContainerCreate(ctx context.Context, config *container.Config
 	err = json.NewDecoder(serverResp.body).Decode(&response)
 	return response, err
 }
+
+// formatPlatform returns a formatted string representing platform (e.g. linux/arm/v7).
+//
+// Similar to containerd's platforms.Format(), but returns an empty string
+// instead of "unknown" if no OS is present: https://github.com/containerd/containerd/blob/v1.5.2/platforms/platforms.go#L243-L263
+func formatPlatform(platform *specs.Platform) string {
+	if platform == nil || platform.OS == "" {
+		return ""
```

Comment:
Let me call out here that containerd's format would return `"unknown"` in this case. As far as I can see, we have no use for the `"unknown"` value, and would treat it equivalent to `""` / not set, so I went for "don't set the parameter if we don't have anything useful". but perhaps I overlooked a specific scenario 👀 👀 👀 👀

Reply:
So, buildkit supports just setting arch... weird that we'd return empty for empty OS.

Reply:
Any thoughts on this?

Reply:
Looked up the parsing on the daemon side, and looks indeed that that should be supported; if only `arch` is specified, the parsing on the daemon side should handle that; https://github.com/moby/moby/blob/ada51d6d29c7517812d5bbfc15614bf505393681/vendor/github.com/containerd/containerd/platforms/platforms.go#L182-L206